### PR TITLE
Scp 1583+1584

### PR DESCRIPTION
**Pull request by dk14**
random walk + api call for testkit

Pre-submit checklist:
- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `$(nix-build default.nix -A pkgsLocal.updateMaterialized)` to update the materialized Nix files
   - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
   - [ ] `$(nix-shell shell.nix --run fix-stylish-haskell)` to fix any formatting issues
- If you changed any Purescript files:
   - [ ] `$(nix-shell shell.nix --run fix-purty)` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: marlowe-actus/src/Language/Marlowe/ACTUS/QCGenerator.hs [hunk 1]
```diff
@@ -9,7 +9,8 @@ import qualified Data.Map                                          as M
 import           Data.Time
 import           Data.Time.Clock.POSIX                             (posixSecondsToUTCTime)
 import           Language.Marlowe.ACTUS.Analysis                   (genProjectedCashflows)
-import           Language.Marlowe.ACTUS.Definitions.BusinessEvents
+import Language.Marlowe.ACTUS.Definitions.BusinessEvents
+    ( RiskFactors(..) )
```
Comment (shmish111) [inline marlowe-actus/src/Language/Marlowe/ACTUS/QCGenerator.hs:13]: strange formatting, has stylish haskell been run?

Comment (dk14) [reply marlowe-actus/src/Language/Marlowe/ACTUS/QCGenerator.hs:13]: Nope. I’m fixing my NixOs VM today coz it took all space on my hard drive

Comment (dk14) [reply marlowe-actus/src/Language/Marlowe/ACTUS/QCGenerator.hs:13]: And then resizing broke it lol

File: marlowe-actus/src/Language/Marlowe/ACTUS/QCGenerator.hs [hunk 2]
```diff
@@ -199,10 +200,22 @@ contractTermsGen = do
...
+
+riskFactorsGenRandomWalk :: ContractTerms -> Gen (M.Map Day RiskFactors)
+riskFactorsGenRandomWalk contractTerms = do
+    rfs <- M.toList <$> riskFactorsGen contractTerms
+    let 
+        fluctuate state fluctiation = state + (fluctiation - 50) / 100
+        walk rf st = 
+            let fluctuate' extractor = fluctuate (extractor rf) (extractor st)
+            in RiskFactors (fluctuate' o_rf_CURS) (fluctuate' o_rf_RRMO) (fluctuate' o_rf_SCMO) (fluctuate' pp_payoff)
+    path <- (\init -> L.scanl walk init (snd <$> rfs)) <$> riskAtT
```
[the remark this answers is not in the document]
Comment (palas) [reply marlowe-actus/src/Language/Marlowe/ACTUS/QCGenerator.hs:220]: If we didn't need `riskFactorsDates` we could avoid the need for `snd` by using `M.elems` instead, and there is also `M.keys` for the opposite case. Just realised that: `zip`, `unzip`, and `scanl` seem to all be in `Prelude` too

**Review by palas — CHANGES_REQUESTED** [review]
Looks good. Just some stylistic comments

**Review by palas — APPROVED** [review]

**Review by shmish111 — APPROVED** [review]
LGTM

**Pull request merged**